packages/workflow/src/Extensions/ArrayExtensions.ts

```diff
@@ -149,7 +145,15 @@ function unique(value: unknown[], extraArgs: string[]): unknown[] {
 	}, []);
 }
 
+const ensureNumberArray = (arr: unknown[], { fnName }: { fnName: string }) => {
+	if (arr.some((i) => typeof i !== 'number')) {
```

Comment:
Nit: Would `isNaN` do the job here?
Also, can we make this a type guard?

Reply:
> Would isNaN do the job here?

Why do you think we should use it? `isNaN` comes with many caveats, e.g. `isNaN("2")` → `false`

> Also, can we make this a type guard?

A type guard would require checking every element, instead of just bailing as soon as we find a non-number. Is the type information worth it?

Reply:
We could do sth like this but this would require changing the signature of the extension for when the type guard is false, or throwing a second error that is never reached to avoid changing the signature, in both cases adding indirection that might not be worth the DX gain.

```ts
const ensureNumberArray = (arr: unknown[], { fnName }: { fnName: string }): arr is number[] => {
	const result = arr.some((i) => typeof i !== 'number');

	if (!result) {
		throw new ExpressionExtensionError(`${fnName}(): all array elements must be numbers`);
	}

	return result;
};

function sum(value: unknown[]): number {
	// Function lacks ending return statement and return type does not include 'undefined'.ts(2366)
	if (ensureNumberArray(value, { fnName: 'sum' })) {
		return value.reduce((p: number, c: unknown) => {
			if (typeof c === 'string') {
				return p + parseFloat(c);
			}
			if (typeof c !== 'number') {
				return NaN;
			}
			return p + c;
		}, 0);
	}
}
```

Reply:
Agree, ideally I would like that it's not using so many `unkown`s but that's out of scope now.